access-grant/src/main/java/com/inrupt/client/accessgrant/AccessGrantClient.java

```diff
@@ -200,6 +200,63 @@ public CompletionStage<AccessGrant> issue(final URI type, final URI agent, final
         });
     }
 
+    /**
+     * A POST Request to be used for issuing an access grant or request.
+     *
+     * @param type the credential type
+     * @param agent the receiving agent for this credential
+     * @param resources the resources to which this credential applies
+     * @param modes the access modes for this credential
+     * @param purposes the purposes of this credential
+     * @param expiration the expiration time of this credential
+     * @return an AccessRequest for issuing an access grant or request
+     */
+    public AccessRequest issueAccessRequest(final URI type, final URI agent, final Set<URI> resources,
+        final Set<String> modes, final Set<String> purposes, final Instant expiration) {
+        return v1Metadata().thenApply(metadata -> {
+            final Map<String, Object> data;
+            if (ACCESS_GRANT.equals(type)) {
+                data = buildAccessGrantv1(agent, resources, modes, expiration, purposes);
+            } else if (ACCESS_REQUEST.equals(type)) {
+                data = buildAccessRequestv1(agent, resources, modes, expiration, purposes);
+            } else {
+                throw new AccessGrantException("Unsupported grant type: " + type);
+            }
+
+            return new AccessRequest(data);
+        }).toCompletableFuture().join();
+    }
+
+    /**
+     * Approves an access grant or request.
+     *
+     * @param request an AccessRequest
+     * @return the next stage of completion containing the resulting credential
+     */
+    public CompletionStage<AccessGrant> approveAccessRequest(final AccessRequest request) {
```

Comment:
I would call this `issueAccessGrant`

Reply:
issueAccessRequest name is to specify it is a request and not the actual issue call. 
I call it better issueGrantRequest